### PR TITLE
refactor(db): change clients.secret to clients.hashedSecret, remove clients.whitelisted

### LINKED ISSUE #155: add clients.hashedSecret column

### LINKED ISSUE #267: [meta] drop the whitelisted column in clients table and associated code and config
this will affect fxa-dev, stage and prod dbs, fxa-oauth-console, others.

### PR DESCRIPTION
**Pull request by seanmonstar**
Closes #155
Closes #267

### CODE REVIEW

File: lib/db/mysql/index.js [hunk 1]
```diff
@@ -299,7 +300,7 @@ MysqlStore.prototype = {
     if (!client.id) {
       return P.reject(new Error('Update client needs an id'));
     }
-    var secret = client.hashedSecret || client.secret || null;
+    var secret = client.hashedSecret;
```
Comment (pdehaan) [inline lib/db/mysql/index.js]: Did you want to rename this to `var hashedSecret = ...` too? (as well as lines 304, 305, and 311)

**Comment by pdehaan** [comment]
Re: @rfk's https://github.com/mozilla/fxa-oauth-server/issues/155#issuecomment-103689026,

> _"Let's do this at the same time as we remove "whitelisted" column, in a train or two's time"_

Did we want to murder the `whitelisted` column in [/lib/db/mysql/index.js:219-220](
https://github.com/mozilla/fxa-oauth-server/blob/7d8aa6d67d44d41784c9a674a519a5f442c13685/lib/db/mysql/index.js#L219-L220) while we're here?

```
!!client.trusted,  // XXX TODO: we have duplicate columns while we're
!!client.trusted,  // in the process of renaming whitelisted=>trusted.
```

**Comment by pdehaan** [comment]
Travis says:

```
fxa-oauth-server.db.ERROR: registerClient Error: ER_BAD_FIELD_ERROR: Unknown column 'hashedSecret' in 'field list'
    at Query.Sequence._packetToError (/home/travis/build/mozilla/fxa-oauth-server/node_modules/mysql/lib/protocol/sequences/Sequence.js:48:14)
    at Query.ErrorPacket (/home/travis/build/mozilla/fxa-oauth-server/node_modules/mysql/lib/protocol/sequences/Query.js:83:18)
    at Protocol._parsePacket (/home/travis/build/mozilla/fxa-oauth-server/node_modules/mysql/lib/protocol/Protocol.js:271:23)
    at Parser.write (/home/travis/build/mozilla/fxa-oauth-server/node_modules/mysql/lib/protocol/Parser.js:77:12)
    at Protocol.write (/home/travis/build/mozilla/fxa-oauth-server/node_modules/mysql/lib/protocol/Protocol.js:39:16)
    at Socket.<anonymous> (/home/travis/build/mozilla/fxa-oauth-server/node_modules/mysql/lib/Connection.js:96:28)
    at Socket.emit (events.js:95:17)
    at Socket.<anonymous> (_stream_readable.js:765:14)
    at Socket.emit (events.js:92:17)
    at emitReadable_ (_stream_readable.js:427:10)
    at emitReadable (_stream_readable.js:423:5)
    at readableAddChunk (_stream_readable.js:166:9)
    at Socket.Readable.push (_stream_readable.js:128:10)
    at TCP.onread (net.js:529:21)
    --------------------
    at Pool.query (/home/travis/build/mozilla/fxa-oauth-server/node_modules/mysql/lib/Pool.js:180:23)
    at /home/travis/build/mozilla/fxa-oauth-server/lib/db/mysql/index.js:1394:12
    at tryCatcher (/home/travis/build/mozilla/fxa-oauth-server/node_modules/bluebird/js/main/util.js:24:31)
    at Promise._resolveFromResolver (/home/travis/build/mozilla/fxa-oauth-server/node_modules/bluebird/js/main/promise.js:427:31)
    at new Promise (/home/travis/build/mozilla/fxa-oauth-server/node_modules/bluebird/js/main/promise.js:53:37)
    at Object._query (/home/travis/build/mozilla/fxa-oauth-server/lib/db/mysql/index.js:1392:8)
    at Object._write (/home/travis/build/mozilla/fxa-oauth-server/lib/db/mysql/index.js:1352:13)
    at Object.registerClient (/home/travis/build/mozilla/fxa-oauth-server/lib/db/mysql/index.js:1021:13)
    at /home/travis/build/mozilla/fxa-oauth-server/lib/db/index.js:472:35
    at tryCatcher (/home/travis/build/mozilla/fxa-oauth-server/node_modules/bluebird/js/main/util.js:24:31)
    at Promise._settlePromiseFromHandler (/home/travis/build/mozilla/fxa-oauth-server/node_modules/bluebird/js/main/promise.js:454:31)
    at Promise._settlePromiseAt (/home/travis/build/mozilla/fxa-oauth-server/node_modules/bluebird/js/main/promise.js:530:18)
    at Promise._settlePromiseAtPostResolution (/home/travis/build/mozilla/fxa-oauth-server/node_modules/bluebird/js/main/promise.js:224:10)
    at Promise.b (domain.js:183:18)
    at Async._drainQueue (/home/travis/build/mozilla/fxa-oauth-server/node_modules/bluebird/js/main/async.js:182:12)
    at Async._drainQueues (/home/travis/build/mozilla/fxa-oauth-server/node_modules/bluebird/js/main/async.js:187:10)
```

**Comment by rfk** [comment]
> Did we want to politely and delicately remove the whitelisted column

Yes, please do, the longer we leave it the more likely we are to forget about ever doing it.

**Comment by pdehaan** [comment]
Side note, it looks like [almost] all the TODOs in this repo are pretty much around the "polite and delicate" removal of the legacy "whitelisted" property. :+1: 

```sh
$ grep -irn "TODO" . --exclude-dir={node_modules,coverage} --exclude=coverage.html

./lib/db/mysql/index.js:218:      !!client.trusted,  // XXX TODO: we have duplicate columns while we\'re
./lib/db/mysql/index.js:314:      client.trusted,  // XXX TODO: we have duplicate columns while we\'re
./lib/db/mysql/index.js:452:    // TODO this should be a transaction or stored procedure
./lib/routes/client/list.js:21:    // XXX TODO: a future PR will remove legacy "whitelisted" attr
./lib/routes/client/list.js:43:          // XXX TODO: a future PR will remove legacy "whitelisted" attr
./lib/routes/client/register.js:28:      // XXX TODO: a future PR will remove legacy "whitelisted" property
./lib/routes/client/register.js:43:      // XXX TODO: a future PR will remove legacy "whitelisted" property
./lib/routes/client/register.js:60:      // XXX TODO: a future PR will remove legacy "whitelisted" property.
./lib/routes/client/register.js:94:          // XXX TODO: a future PR will remove legacy "whitelisted" property
./test/api.js:1259:            // XXX TODO: future PR will remove legacy "whitelisted" attr,
```

File: lib/db/mysql/index.js (hunk 1 above)
Comment (seanmonstar) [inline lib/db/mysql/index.js]: I didn't really feel a new here, as they're just local variables, and the intent is clear.

**Comment by seanmonstar** [comment]
Le sigh, I always forget to increase the `patch.js`.

**Comment by seanmonstar** [comment]
Good to review.

**Comment by rfk** [comment]
r+; but let's make sure we've got everything updated to not use "whitelisted".  For example: https://github.com/mozilla/fxa-oauth-client/pull/6

**Pull request merged**